content/docs/rfcs/26/README.md

```diff
@@ -0,0 +1,88 @@
+---
+slug: 26
+title: 26/WAKU-PAYLOAD
+name: Waku Message Payload Encryption
+status: draft
+editor: Oskar Thoren <oskar@status.im>
+contributors:
+---
+
+This specification describes how encryption, decryption and signing works in
+[6/WAKU1](/spec/6) and in [10/WAKU2](/spec/10) with [14/WAKU-MESSAGE version
+1](/spec/14/#version1).
+
+It effectively replaces [7/WAKU-DATA](/spec/7) as well as [6/WAKU1 Payload
+encryption](/spec/6/#payload-encryption) but written in a way that is agnostic
+and self-contained for Waku v1 and Waku v2.
+
+## Cryptographic primitives
+
+- AES-256-GCM
+- ECIES
+- ECSDA
+- KECCAK-256
+
+## Payload encryption
+
+Asymmetric encryption uses the standard Elliptic Curve Integrated Encryption Scheme (ECIES) with SECP-256k1 public key.
+In case of a signature being provided, the public key is recoverable from by utilizing the `v` parameter.
+
+Symmetric encryption uses AES-256-GCM for authenticated encryption [TODO], with a 16 byte authentication tag 16 and a 12 byte IV (nonce).
+
+## Specification
+
+For 6/WAKU1, the `data` field is used within the `waku envelope`, the field MUST contain the encrypted payload of the `waku envelope`.
+
+For 10/WAKU2, the `payload` field is used within the `WakuMessage` and MUST contain the encrypted payload of `Waku Message`.
+
+The fields that are concatenated and encrypted as part of the field are:
+ - flags
+ - auxiliary field
+ - payload
+ - padding
+ - signature
+ 
+In case of symmetric encryption, a `salt`  (a.k.a. AES Nonce, 12 bytes) field MUST be appended. 
```

Comment:
We are using 3 terminology (salt, iv, nonce) to refer to the same thing. Maybe best to reduce to 1 or at least 2? (e.g, IV + Nonce).

Reply:
Yeah agree, usually IV or nonce is used for this in literature afaict, somehow salt was used in early Whisper specs, which usually goes with passwords afaik. (Haven't gotten to this section yet)